### PR TITLE
Prefer filename as title over "Untitled"

### PR DESCRIPTION
**Pull request by roggan87**
Extract filename and use as html title if the <title> tag isn't present. Unfortunately I haven't tried the code since I can't set up the correct compilation environment, but it should work (more or less).

### CODE REVIEW

**Comment by roggan87** [comment]
Sorry, line 366 should read:

    var docName = svgRoot[0].firstChild.getAttributeNode("sodipodi:docname").value.trim();

I forgot to add [0] after svgRoot.

File: js/model/Presentation.js [hunk 1]
```diff
@@ -357,7 +357,17 @@ export var Presentation = {
 
     get title() {
         var svgTitles = this.document.root.getElementsByTagNameNS(SVG_NS, "title");
-        return svgTitles.length ? svgTitles[0].firstChild.wholeText.trim() : "Untitled";
+        if (svgTitles.length) {
+            return svgTitles[0].firstChild.wholeText.trim();
+        }
+        else {
+            var svgRoot = this.document.root.getElementsByTagNameNS(SVG_NS, "svg");
```
Comment (senshu) [inline js/model/Presentation.js]: Actually, `this.document.root` is the root SVG element of the presentation.

File: js/model/Presentation.js [hunk 2]
```diff
@@ -357,7 +357,17 @@ export var Presentation = {
...
+            if (svgRoot.length) {
+                var docName = svgRoot.firstChild.getAttributeNode("sodipodi:docname").value.trim();
```
Comment (senshu) [inline js/model/Presentation.js]: This part assumes that the document was generated with Inkscape.
We should return "Untitled" if the attribute `sodipodi:docname` does not exist.

File: js/model/Presentation.js (hunk 1 above)
Comment (roggan87) [inline js/model/Presentation.js]: Thanks for pointing out. I just got my environment up and running so now I'm able to try out the code for real. Will look a bit more on this pull request and fine tune.

File: js/model/Presentation.js (hunk 2 above)
Comment (roggan87) [inline js/model/Presentation.js]: Oh I see. Maybe it's better to parse the file path that is used internally by sozi to refer to the file being opened?

Line 370 will return "Untitled" if the element attribute does not exist. (at least in theory)

Comment (senshu) [inline js/model/Presentation.js]: > Line 370 will return "Untitled" if the element attribute does not exist. (at least in theory)

As I understand it, line 370 is executed if `svgTitles.length` and `svgRoot.length` are both equal to zero.

Comment (roggan87) [inline js/model/Presentation.js]: You're correct. I'll take a closer look at this and make a new pull request when it's a bit more polished.

**Comment by roggan87** [comment]
Incomplete, will fine tune and make a new pull request.

**Pull request closed without merging**